nvm.sh

```diff
@@ -2724,26 +2727,25 @@ nvm() {
     return
   fi
 
-  local DEFAULT_IFS
-  DEFAULT_IFS=" $(nvm_echo t | command tr t \\t)
+  local IFS
+  IFS=" $(nvm_echo t | command tr t \\t)
 "
-  if [ "${-#*e}" != "$-" ]; then
-    set +e
-    local EXIT_CODE
-    IFS="${DEFAULT_IFS}" nvm "$@"
-    EXIT_CODE="$?"
-    set -e
-    return "$EXIT_CODE"
-  elif [ "${-#*a}" != "$-" ]; then
-    set +a
+  local OPTIONS=''
+  local O='e a'
+  # set +E only disables errtrace on Bash
+  [ -z "${BASH-}" ] || O='e a E'
+  for O in $O; do
```

Comment:
I’m not sure this will separate properly on zsh?

Reply:
Indeed. (Not without `setopt shwordsplit`, anyway--but that would be something else to reverse afterwards.) Portable shell scripting is great fun, isn't it?

For now I'll just revert all of this to the original syntax and add `-E` in the same way.

It might be worth avoiding all of these shenanigans by having nvm run (mostly) in a subshell and output a script that is then `eval`'d to update environment variables, but that's a more significant undertaking, obviously.